src/switch_core_cert.c

```diff
@@ -69,7 +70,7 @@ SWITCH_DECLARE(void) switch_ssl_init_ssl_locks(void)
 			switch_assert(ssl_mutexes[i] != NULL);
 		}
 
-		CRYPTO_set_id_callback(switch_ssl_ssl_thread_id);
+		CRYPTO_THREADID_set_callback(switch_ssl_ssl_thread_id);
```

Comment:
From  open ssl 1.1.0, this two MACROs ard defined as empty. So the function switch_ssl_ssl_thread_id and switch_ssl_ssl_lock_callback will not be used and will trigger the W-unused-function errors in Clang. (Gcc will ignore this warning for static inline, but clang will not). Then Clang build will fail. Could you also fix this problem in your PR?

Reply:
if that is actually unused does this mean me should be doing something differently here ?

Reply:
> From open ssl 1.1.0, this two MACROs ard defined as empty. So the function switch_ssl_ssl_thread_id and switch_ssl_ssl_lock_callback will not be used and will trigger the W-unused-function errors in Clang. (Gcc will ignore this warning for static inline, but clang will not). Then Clang build will fail. Could you also fix this problem in your PR?

Hi again, I looked at it but it was over my head. These functions/macros are used in various places and if they are ifdef'ed that should probably include other functions as well (CRYPTO_set_locking_callback for instance?).